p2p/transport/pipe/pipetransport.go

```diff
@@ -0,0 +1,93 @@
+package pipetransport
+
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"sync"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+	tpt "github.com/libp2p/go-libp2p-transport"
+	tptu "github.com/libp2p/go-libp2p-transport-upgrader"
+	ma "github.com/multiformats/go-multiaddr"
+)
+
+type PipeTransport struct {
+	mlistenchans *sync.RWMutex
+	listenchans  map[string]chan *PipeConn
```

Comment:
do we really care to support multiple connections for this?

Reply:
Even if that not a thing here I think we can expect some people or other maybe dialling multiple process via shared memory or multiple instance of libp2p in a same process for debugging other layer of libp2p in the future.